modules/apps/bookmarks/bookmarks-web/src/main/resources/META-INF/resources/bookmarks/js/BookmarksManagementToolbarPropsTransformer.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+const ACTIONS = {
+	deleteEntries({
+		deleteEntriesURL,
+		inputId,
+		inputValue,
+		portletNamespace,
+		trashEnabled,
+	}) {
+		if (
+			trashEnabled ||
+			confirm(
+				Liferay.Language.get(
+					'are-you-sure-you-want-to-delete-the-selected-entries'
+				)
+			)
+		) {
+			const form = document.getElementById(`${portletNamespace}fm`);
+
+			form.setAttribute('method', 'post');
+
+			const inputElement = document.getElementById(
+				`${portletNamespace}${inputId}`
+			);
+
+			if (inputElement) {
+				inputElement.setAttribute('value', inputValue);
+
+				submitForm(form, deleteEntriesURL);
+			}
+		}
+	},
+};
+
+export default function propsTransformer({
+	additionalProps,
+	initialActionDropdownItems,
+	portletNamespace,
+	...props
+}) {
+	const processAction = (action) => {
+		const {
+			deleteEntriesURL,
+			inputId,
+			inputValue,
+			trashEnabled,
+		} = additionalProps;
+
+		ACTIONS[action]({
+			deleteEntriesURL,
+			inputId,
+			inputValue,
+			portletNamespace,
+			trashEnabled,
+		});
+	};
+
+	return {
+		...props,
+		initialActionDropdownItems: initialActionDropdownItems.map((item) => {
+			return {
+				...item,
+				onClick() {
+					const action = item.data.action;
```

Comment:
It should be `item.data?.action`, see https://github.com/liferay-lima/liferay-portal/pull/1467#discussion_r571028315